src/client/entity-editor/series-section/actions.ts

```diff
@@ -81,6 +86,52 @@ export function addSeriesItem(data: Relationship): Action {
 	};
 }
 
+
+/**
+ * This action creator first sorts the series items object and then pass the
+ * sorted object in the payload while dispatching the action.
+ *
+ * @param {number} oldIndex - Old Position of the series item.
+ * @param {number} newIndex - New Position of the series item.
+ * @returns {void}
+ */
+export function sortSeriesItems(oldIndex, newIndex):any {
+	return (dispatch, getState) => {
+		const state = getState();
+		const seriesItems = state.get('seriesSection').get('seriesItems');
+		const orderTypeValue = state.get('seriesSection').get('orderType');
+		const seriesItemsObject = seriesItems.toJS();
+		const seriesItemsArray = Object.entries(seriesItemsObject);
```

Comment:
It seems to me a bit cumbersome to have to use entries and end up with [ [string , Relationship] ] and having to get the relationship properties with `seriesItem[1].attributes`, instead of using Object.values() and getting an array of Relationship objects directly (considering the rowID is also in the Relationship object).
Does it seem easier to you as well to work with a Relationship[] instead?

I do see that further down you do `Object.fromEntries(new Map([...sortedSeriesItems]))`, maybe that's the reason you opted for this?
In that case, perhaps a simple one-liner array.reduce can give us the same result:
`const payload = sortedSeriesItems.reduce((accumulator,rel:RelationshipForDisplay)=> (accumulator[rel.rowID]=rel,accumulator),{});`

I'm using the type RelationshipForDisplay instead of Relationship as it has the rowID defined.

Reply:
Yeah looks good, I will refactor it :)